langserver/langserver.py

```diff
@@ -50,17 +54,17 @@ def handle(self, request):
             "exit": self.serve_exit,
         }.get(request["method"], self.serve_default)
 
-        # We handle notifs differently since we can't respond
+        # We handle notifications differently since we can't respond
         if "id" not in request:
             try:
-                handler(request)
+                handler(request, handler_span)
```

Comment:
passing in `handler_span` everywhere seems annoying and we can make it more convenient in python. Most web request frameworks have the concept of a request context. I'd be happy to just add another field to request called `span`. That should make the code clearer without affecting every call site.